lib/ansible/cli/__init__.py

```diff
@@ -192,10 +192,15 @@ def split_vault_id(vault_id):
         ret = tuple(parts)
         return ret
 
+    # TODO: remove `create_new_password` as a parameter once the deprecation
+    # time has passed.
     @staticmethod
     def build_vault_ids(vault_ids, vault_password_files=None,
                         ask_vault_pass=None, create_new_password=None,
                         auto_prompt=True):
+        if create_new_password is not None:
+            display.deprecated("create_new_password is not used in build_vault_ids anymore and is only being kept to maintain compatibility,"
+                               " expect it to be removed in a future release.", version="2.19")
```

Comment:
If there's a third party caller passing `create_new_password` as None, they won't get this deprecation warning, but they will get a TypeError when we delete the kwarg. Here's one way to fix that:

```diff
+def _handle_deprecated_kwarg(func):
+    def wrapper(*args, **kwargs):
+        if len(args) >= 4 or 'create_new_password' in kwargs:
+            display.deprecated("msg", version='2.19')
+        return func(*args, **kwargs)
+    return wrapper
+
+
 class CLI(ABC):
     ''' code behind bin/ansible* programs '''
 
@@ -192,6 +200,7 @@ class CLI(ABC):
         ret = tuple(parts)
         return ret
 
+    @_handle_deprecated_kwarg
     @staticmethod
     def build_vault_ids(vault_ids, vault_password_files=None,
                         ask_vault_pass=None, create_new_password=None,
```

> create_new_password is not used in build_vault_ids anymore ...

It was never used, so this should just indicate that it's unused and will be removed. No need to mention backwards compatibility for private APIs.

Reply:
Thanks for the code! Hopefully the deprecation notice will now display if someone *is* using `create_new_password` by passing in `None`.

Reply:
Thanks for the fix! Added you as a co-author to this commit for providing the code for the fix and suggesting a proper deprecation notice.